politeiawww/cmd/politeiavoter/politeiavoter.go

```diff
@@ -554,86 +477,127 @@ func (c *ctx) eligibleVotes(vrr *v1.VoteResultsReply, ctres *pb.CommittedTickets
+// returns the reply.
+func (c *ctx) voteDetails(token, serverPubKey string) (*tkv1.DetailsReply, error) {
+	d := tkv1.Details{
+		Token: token,
+	}
+	responseBody, err := c.makeRequest(http.MethodPost,
+		tkv1.APIRoute, tkv1.RouteDetails, d)
 	if err != nil {
-		return err
+		return nil, err
 	}
 
-	// Get latest block
-	ar, err := c.wallet.Accounts(c.wctx, &pb.AccountsRequest{})
+	var dr tkv1.DetailsReply
+	err = json.Unmarshal(responseBody, &dr)
 	if err != nil {
-		return err
+		return nil, fmt.Errorf("Could not unmarshal DetailsReply: %v",
+			err)
 	}
-	latestBlock := ar.CurrentBlockHeight
-	//fmt.Printf("Current block: %v\n", latestBlock)
 
-	for _, v := range i.Votes {
-		// Make sure we have a CensorshipRecord
-		if v.Proposal.CensorshipRecord.Token == "" {
-			// This should not happen
-			log.Debugf("skipping empty CensorshipRecord")
-			continue
-		}
+	// Verify VoteDetails.
+	err = client.VoteDetailsVerify(*dr.Vote, serverPubKey)
+	if err != nil {
+		return nil, err
+	}
 
-		// Make sure we have valid vote bits
-		if v.StartVote.Vote.Token == "" || v.StartVote.Vote.Mask == 0 ||
-			v.StartVote.Vote.Options == nil {
-			// This should not happen
-			log.Errorf("invalid vote bits: %v",
-				v.Proposal.CensorshipRecord.Token)
-			continue
+	return &dr, nil
+}
+
+func (c *ctx) voteResults(token, serverPubKey string) (*tkv1.ResultsReply, error) {
+	r := tkv1.Results{
+		Token: token,
+	}
+	responseBody, err := c.makeRequest(http.MethodPost,
+		tkv1.APIRoute, tkv1.RouteResults, r)
+	if err != nil {
+		return nil, err
+	}
+
+	var rr tkv1.ResultsReply
+	err = json.Unmarshal(responseBody, &rr)
+	if err != nil {
+		return nil, fmt.Errorf("Could not unmarshal ResultsReply: %v", err)
+	}
+
+	// Verify CastVoteDetails.
+	for _, cvd := range rr.Votes {
+		err = client.CastVoteDetailsVerify(cvd, serverPubKey)
+		if err != nil {
+			return nil, err
 		}
+	}
+
+	return &rr, nil
+}
 
-		// Sanity, check if vote has expired
-		endHeight, err := strconv.ParseInt(v.StartVoteReply.EndHeight, 10, 32)
+func (c *ctx) inventory() error {
+	// Get server public key to verify replies.
+	version, err := c.getVersion()
+	if err != nil {
+		return err
+	}
+	serverPubKey := version.PubKey
+	// Inventory route is paginated, therefore we keep fetching
+	// until we receive a patch with number of records smaller than the
+	// ticketvote's declared page size.
+	page := uint32(1)
+	var tokens []string
+	for {
+		ir, err := c._inventory(tkv1.Inventory{
+			Page:   page,
+			Status: tkv1.VoteStatusStarted,
+		})
 		if err != nil {
 			return err
 		}
-		if int64(latestBlock) > endHeight {
-			// Should not happen
-			fmt.Printf("Vote expired: current %v > end %v %v\n",
-				endHeight, latestBlock, v.StartVote.Vote.Token)
-			continue
+		pageTokens := ir.Vetted[tkv1.VoteStatuses[tkv1.VoteStatusStarted]]
+		tokens = append(tokens, pageTokens...)
+		if uint32(len(pageTokens)) < tkv1.InventoryPageSize {
+			break
+		}
+		page++
+	}
+
```

Comment:
If there are no active votes then politeiavoter just exits without printing anything.  We should print some type of message to let the user know that there are no active votes.

Reply:
```
$ politeiavoter inventory --politeiawww "https://localhost:4443" --skipverify   
No active votes found.
```